src/policy-store.ts

```diff
@@ -293,6 +293,32 @@ export class PolicyStore extends PolicyStoreBase {
     return buildSchema(namespace, actionNames, groupEntityTypeName);
   }
 
+  /**
+   * This method generates a schema based on an AWS CDK RestApi construct. It makes the same assumptions
+   * and decisions made in the Amazon Verified Permissions console.
+   *
+   * @param restApi The RestApi construct instance from which to generate the schema.
+   * @param groupEntityTypeName Specifies a group entity type name. If passed, the schema's User type will have a parent of this type.
+   */
+  public static schemaFromRestApi(restApi: RestApi, groupEntityTypeName?: string) {
+    const namespace = cleanUpApiNameForNamespace(restApi.restApiName);
+    const actionNames: string[] = [];
+    for (const method of restApi.methods) {
+      const pathVerb = method.httpMethod.toLowerCase();
+      const pathUrl = method.resource.path;
+      if (pathVerb === 'any') {
+        for (const verb of this.RELEVANT_HTTP_METHODS) {
+          actionNames.push(`${verb} ${pathUrl}`);
+        }
+      }
+	  if (this.RELEVANT_HTTP_METHODS.includes(pathVerb)) {
+      	actionNames.push(`${pathVerb} ${pathUrl}`);
+	  }
+    }
+    return buildSchema(namespace, actionNames, groupEntityTypeName);
+  }
+
+  private static RELEVANT_HTTP_METHODS = ['get', 'post', 'put', 'patch', 'delete', 'head'];
```

Comment:
I would not keep it as static, a const should be fine. So we avoid problems with `this` accessibility. 
See similar example: [Playground Link](https://www.typescriptlang.org/play/?#code/MYGwhgzhAEAqCmEAu0DeBYAUNH0AOATgJYBuYS80y5Rw0ASgKIAyjAagIIBysA+gBKxYABV4BZRrH4B5ACIBlaAF5oAbQDkAc3hJ1AGmjq8Ae2T7DeAK66DR8sAAW59QBN4IHfGcP4YF+oBdAG4sLFx8SwAjEFoqJBo6CmQxHQdjFwAKAEo0MPDcYGMAOwhjDwA6EGNNDKQHIghyplZOHgEhUQkpOXlVAAYArJDsXABfLHHMQpKy+ErqjIAiWHqYCDTLEBdoAHdjAgBrRaGsBGRypKQUuvTs4enSiqqa5dWqDa3oIuMUPcPjrBkAjQSLKOCIJAXCHXNIuYaRO5AA)

Reply:
Just refactored this. Thanks for the speedy review!